tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -184,15 +187,19 @@ def get_metric_data(self, resource):
         cached_metric_data = self._get_cached_metric_data(resource)
         if cached_metric_data:
             return cached_metric_data['measurement']
-
-        metrics_data = self.client.metrics.list(
-            resource['id'],
-            timespan=self.timespan,
-            interval=self.interval,
-            metricnames=self.metric,
-            aggregation=self.aggregation,
-            filter=self.filter
-        )
+        try:
+            metrics_data = self.client.metrics.list(
+                resource['id'],
+                timespan=self.timespan,
+                interval=self.interval,
+                metricnames=self.metric,
+                aggregation=self.aggregation,
+                filter=self.filter
+            )
+        except self.client.models(self.client._get_api_version('metrics')).ErrorResponseException:
```

Comment:
I think if we do `self.client.models.ErrorResponseException` it gives us a default version

Reply:
I thought that too, but the default versions for metrics is different then the default version for the Monitor SDK as a whole: https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/monitor/azure-mgmt-monitor/azure/mgmt/monitor/_monitor_management_client.py#L72

So when I tried just getting the default it wasn't catching the errors because they were from different versions

Reply:
Yeah this isn't ideal.  Can we just catch the base class instead? `from msrest.exceptions import HttpOperationError` ?